src/smc-webapp/project_store.coffee

```diff
@@ -932,26 +930,99 @@ class ProjectActions extends Actions
                 path            : '.'
                 cb              : (err, output) =>
                     if not err
-                        @setState(library_available: {"#{k}" : (output.exit_code == 0)})
+                        library = @get_store().library
+                        library = library.set(k, (output.exit_code == 0))
+                        @setState(library: library)
                     cb(err)
-        setTimeout(async.eachOfSeries(LIBRARY, check), 1000)
 
-    copy_from_library: (opts) =>
-        opts = defaults opts,
-            entry : 'first_steps'
+        # reads the index file for the library (if it fails, we have no library)
+        index = (cb) =>
+            {webapp_client} = require('./webapp_client')
+
+            webapp_client.read_text_file_from_project
+                project_id : @project_id
+                path       : require('./library').examples_path + '/index.json'
+                cb         : (err, response) =>
+                    if err
+                        cb(); return
+                    try
+                        data = misc.from_json(response.content)
+                        library = @get_store().library.set('examples', data)
```

Comment:
Per comment above, call `immutable.fromJS(data)` here so we don't have a mutability leak.